### PR TITLE
Remove Convert to Blocks button from Legacy Widget in favour of a block transform

### PR DESCRIPTION
**Pull request by noisysocks**
## Description

Alternative to https://github.com/WordPress/gutenberg/pull/32859.

Removes the _Convert to Blocks_ button from the Legacy Widget block in favour of a _Paragraph_ block transform.

Before:

https://user-images.githubusercontent.com/612155/132275148-b612c8d0-04d5-4318-924b-bda16e72f8c5.mp4

After:

https://user-images.githubusercontent.com/612155/132275234-aa170d58-49e7-43f9-a1b0-f71ccb67e075.mp4

Pros of this approach:

- More consistent: This means that converting a Text widget works the same as converting any other type of widget.
- There's a live preview (on hover) of what will happen.

Cons of this approach:

- The transform is labelled "Paragraph" but technically it will transform to images, video, etc.

## How has this been tested?
1. Install the Classic Widgets plugin and insert a Text widget into one of your sidebars.
2. De-activate the Classic Widgets plugin and go to Appearance → Widgets.

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by critterverse** [comment]
>The transform is labelled "Paragraph" but technically it will transform to images, video, etc.

I'm not sure how obvious the action is with this label (whereas "Convert to blocks" is super clear).

Does the item in the Transform menu definitely need to be a block type? Wondering if something like this could work:

transform-to-blocks

File: packages/widgets/src/blocks/legacy-widget/transforms.js [hunk 1]
```diff
@@ -186,7 +186,27 @@ const legacyWidgetTransforms = [
 } );
 
 const transforms = {
-	to: legacyWidgetTransforms,
+	to: [
+		...legacyWidgetTransforms,
+		{
```
Comment (spacedmonkey) [inline packages/widgets/src/blocks/legacy-widget/transforms.js:191]: Why is this added here like this? Can't it just be part of the legacyWidgetTransforms object?

File: packages/widgets/src/blocks/legacy-widget/transforms.js [hunk 2]
```diff
@@ -186,7 +186,27 @@ const legacyWidgetTransforms = [
...
+			type: 'block',
+			blocks: [ 'core/paragraph' ],
+			isMatch( { idBase, instance } ) {
+				return idBase === 'text' && !! instance?.raw;
```
Comment (spacedmonkey) [inline packages/widgets/src/blocks/legacy-widget/transforms.js:195]: Would this also work for html legacy html widgets as well?

**Comment by spacedmonkey** [comment]
This is a looking great! Nearly there. Just a couple of questions and I will approve.

**Comment by noisysocks** [comment]
> Does the item in the Transform menu definitely need to be a block type? 

It's very difficult to implement the mockup you shared with how transforms are currently implemented, but nothing's impossible... 😅

I think I'm still of the mind that this should be consistent with how the Classic block works. So either we remove _Convert to blocks_ from the Legacy Widget and Classic blocks in favour of a Blocks transform **or** we add _Convert to blocks_ to Legacy Widget for all widget types, not just text widgets.

**Comment by draganescu** [comment]
> we add Convert to blocks to Legacy Widget for all widget types, not just text widgets.

I am here to vote for this option. It makes the most sense. _We_ know that we transform a legacy widget, but what we actually do is convert to blocks. So that's what the button should say :)

**Review by spacedmonkey — APPROVED** [review]
I like this PR!

**Comment by gwwar** [comment]
> Does the item in the Transform menu definitely need to be a block type? Wondering if something like this could work:

@critterverse What if we added a shortcut above the transform list? I think ideally it should still say "Convert to blocks". I also think keeping the shortcut in a toolbar, would be ideal to keep discoverability high. I'm guessing part of the UX confusion here is that we have legacy content where the primary action should be transforming it, but instead we see the wildcard transforms from group/columns which are less helpful.

Being able to preview a "Convert to blocks" shortcut as a hover preview would be a nice enhancement for a follow up.

CleanShot 2021-10-11 at 14 12 37@2x

**Comment by critterverse** [comment]
>What if we added a shortcut above the transform list?

Great idea @gwwar. If we go with this approach, maybe this option could be separated with a divider line (similar to how Transforms and Styles are separated when they appear in the same popover):

Transform menu showing convert shortcut at the top of the list in its own section

BTW that squiggly icon in the toolbar actually represents a "Move to widget area" action (but has [been flagged](https://github.com/WordPress/gutenberg/issues/32119) as unhelpful/confusing)!

I think it's an either/or between the Transform menu and the explicit "Convert to blocks" text in the toolbar — with the Transform menu being a tidier UX/UI solution and the toolbar action being much more discoverable. I have a slight preference for keeping the text in the toolbar but either could definitely work.

**Comment by adamziel** [comment]
I like this direction! For what it's worth, it sounds in line with what I discussed with @noisysocks in the other PR: https://github.com/WordPress/gutenberg/pull/35028#discussion_r719959953

**Comment by gwwar** [comment]
> BTW that squiggly icon in the toolbar actually represents a "Move to widget area" action

@critterverse Oh I meant keeping a text "Convert to blocks" shortcut in the toolbar, in addition to the the shortcut in the transform list. I find it's helpful to have actions in multiple places, but I don't have strong feelings on this one.

**Comment by critterverse** [comment]
@gwwar Oh, got it! Thanks for clarifying 😅

Maybe the current text link in the toolbar could be moved into the ellipses menu? That would allow for a couple of different access points but might feel slightly less duplicative. It's also kind of a nice parallel with how grouping works (as an option in both the Transform and ellipses menus).

**Comment by noisysocks** [comment]
Thanks all. I'm going to close out the three PRs we have that explored this and turn https://github.com/WordPress/gutenberg/pull/34610#issuecomment-941038319 into an issue since that's the direction we'd like to head next.
